src/components/asthma/helpers/asthma-data.ts

```diff
@@ -0,0 +1,338 @@
+        ];
+    },
+    loadAirQualitiesForDate: async function (date: Date): Promise<AsthmaAirQuality[]> {
+        let result = await loadAirQualities(startOfDay(date), endOfDay(date));
+        const computeAqi = (dataPoints: DeviceDataPoint[]) => {
+            let maxValue = 0;
+            dataPoints.forEach(dataPoint => {
+                let value = getDataPointValueAsNumber(dataPoint);
+                if (value > maxValue) {
+                    maxValue = value;
+                }
+            });
+            return maxValue;
+        };
+        return [
+            computeAirQuality('home', result.deviceDataPoints, computeAqi, 'not-found'),
+            computeAirQuality('work', result.deviceDataPoints, computeAqi, 'not-found')
+        ];
+    },
+    saveLogEntry: function (logEntry: AsthmaLogEntry): Promise<void> {
+        let logEntryDataPoint: PersistableDeviceDataPoint = {
+            type: 'Asthma-LogEntry',
+            identifier: logEntry.identifier,
+            observationDate: `${logEntry.identifier}T12:00:00-06:00`,
+            value: JSON.stringify(logEntry)
+        };
+        return MyDataHelps.persistDeviceData([logEntryDataPoint]);
+    },
+    loadAndClearAlertTakeover: async function (): Promise<string | undefined> {
+        let today = startOfToday();
+
+        let params: DeviceDataPointQuery = {
+            namespace: 'Project',
+            type: ['AlertTakeover'],
+            observedAfter: today.toISOString()
+        };
+
+        let result = await MyDataHelps.queryDeviceData(params);
+
+        let dataPoints = result.deviceDataPoints;
+        if (dataPoints.length === 0) return undefined;
+
+        let mostRecentDataPoint = dataPoints.sort(sortDataPointsDesc)[0];
+        if (mostRecentDataPoint.properties?.hasOwnProperty('viewed')) return undefined;
+
+        mostRecentDataPoint.properties = mostRecentDataPoint.properties ?? {};
+        mostRecentDataPoint.properties['viewed'] = formatISO(new Date());
+        MyDataHelps.persistDeviceData([mostRecentDataPoint]).then();
+
+        return mostRecentDataPoint.value;
+    },
+    loadSurveyAnswers: async function (surveyNames: string[]): Promise<SurveyAnswer[]> {
+        let answers: SurveyAnswer[] = [];
+
+        let queryParameters: SurveyAnswersQuery = {surveyName: surveyNames};
+
+        const getPage = async function (pageID?: Guid): Promise<SurveyAnswer[]> {
+            queryParameters.pageID = pageID;
+
+            let response = await MyDataHelps.querySurveyAnswers(queryParameters);
+
+            answers = answers.concat(response.surveyAnswers);
+            if (response.nextPageID) {
+                return getPage(response.nextPageID);
+            } else {
+                return answers;
+            }
+        };
+
+        return getPage().then(function (surveyAnswers) {
+            return surveyAnswers;
+        });
+    },
+    loadAsthmaActionPlan: async function (): Promise<AsthmaActionPlan | undefined> {
```

Comment:
Hmm...wondering if this is necessary...this ends up being a report on the back-end right?  Can we just get the report ID and reference it?  Or does it need to be base64 encoded for some reason

Reply:
Yeah, I think given we have a pending refactor of the image stuff it probably doesn't matter too much if this works for now.